frontend/packages/topology/src/components/quick-search/QuickSearchDetails.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+import { useTranslation } from 'react-i18next';
+import { history, removeQueryArgument } from '@console/internal/components/utils';
+import { CatalogItem } from '@console/plugin-sdk';
+import {
+  Button,
+  ButtonVariant,
+  Text,
+  TextContent,
+  TextVariants,
+  Title,
+} from '@patternfly/react-core';
+import './QuickSearchDetails.scss';
+
+interface QuickSearchDetailsProps {
+  selectedItem: CatalogItem;
+}
+
+const QuickSearchDetails: React.FC<QuickSearchDetailsProps> = ({ selectedItem }) => {
+  const { t } = useTranslation();
+
+  return (
+    <div className="odc-quick-search-details">
+      <Title headingLevel="h4">{selectedItem?.name}</Title>
+      {selectedItem?.provider && (
+        <span className="odc-quick-search-details__provider">
+          {t('topology~Provided by')} {selectedItem.provider}
+        </span>
+      )}
+      <Button
+        variant={ButtonVariant.primary}
+        className="odc-quick-search-details__form-button"
+        onClick={() => {
+          removeQueryArgument('catalogSearch');
+          history.push(selectedItem?.cta?.href);
+        }}
+      >
+        {selectedItem?.cta?.label}
+      </Button>
+      <TextContent>
+        <Text component={TextVariants.p} className="odc-quick-search-details__description">
+          {selectedItem?.description}
+        </Text>
+      </TextContent>
+    </div>
```

Comment:
This component kinda falls over if it gets partial / invalid data... This is not good. I think we may need an error state, or we just put an alert saying we had troubles reading the type. 

Right now, it's going to be awkwardly empty... with a Button "shell" that has no label.

Reply:
I didnt came across that scenario yet but it seems like a possibility. We can have have state that says `Data not available for this item` or something similar. cc: @bgliwa01

Reply:
You can rely on the API of `CatalogItem` to be populated correctly.

Reply:
Based on Christian's comment, do we need to worry about this use case? I don't think we have this scenario in the developer catalog so I'm not sure how it would happen here.

Reply:
Do we need to worry about it from a user point of view? No, probably not... but it is a concern right now from a gap in the code. This is how bugs happen. It's probably an edge case, but I'd prefer not to see an empty card with a broken looking button if something goes off the rail.

Reply:
@andrewballantyne we dont need an empty state since this comp will always have a `selectedItem` and if some info is missing from the item itself then we will not show that particular info. Did you came across any scenario where you saw an empty details section?

Reply:
Sorry, looking at a stale snippet above... you removed all optional conditions. My comments are moot now. Thanks!